src/ApiCore/GapicClientTrait.php

```diff
@@ -160,8 +170,30 @@ private function buildClientOptions(array $options)
         $options['transportConfig'] += $defaultOptions['transportConfig'];
         $options['transportConfig']['grpc'] += $defaultOptions['transportConfig']['grpc'];
         $options['transportConfig']['rest'] += $defaultOptions['transportConfig']['rest'];
-
+        
         $this->modifyClientOptions($options);
+
+        if (isset($options['gcpApiConfigPath'])
```

Comment:
Actually, one more small note: WDYT about adding a check for semaphores here before enabling the optimizer, like in [the PR on google-cloud-php](https://github.com/GoogleCloudPlatform/google-cloud-php/pull/1146/files#diff-a759887e9614192a96db41b91bc02147R173)?

Reply:
done. Nice find!

Reply:
Thanks! :)

Do you think we might need to update the test case a bit now too? I have a suspicion if someone running the tests doesn't have semaphores support installed they may see failures.

Reply:
Thank you David. Actually I'm not quite familiar with this, can you tell me a bit more on how to test this? Do we want to create a test environment with that extension being disabled? Or just simply mock the flag to be false?

Reply:
I would be satisfied for now with marking the test as skipped if the extension isn't available.

```php
if (!extension_loaded('sysvshm')) {
    $this->markTestSkipped('The sysvshm extension must be installed to execute this test.');
}
```

WDYT?

Reply:
done.